src/data/nat/factorization.lean

```diff
@@ -42,7 +43,30 @@ namespace nat
 
 /-- `n.factorization` is the finitely supported function `ℕ →₀ ℕ`
  mapping each prime factor of `n` to its multiplicity in `n`. -/
-noncomputable def factorization (n : ℕ) : ℕ →₀ ℕ := (n.factors : multiset ℕ).to_finsupp
+def factorization (n : ℕ) : ℕ →₀ ℕ :=
+{ support := n.factors.to_finset,
+  to_fun := λ p, if p.prime then padic_val_nat p n else 0,
+  mem_support_to_fun :=
+    begin
+      intro a,
+      rw list.mem_to_finset,
+      by_cases hn0 : n = 0,
+      { sorry, },
+      rw mem_factors hn0,
+      simp only [ne.def, ite_eq_right_iff, exists_prop],
+      sorry,
+    end }
+
+
+lemma multiplicity_eq_factorization {n p : ℕ} (pp : p.prime) (hn : n ≠ 0) :
+  multiplicity p n = n.factorization p :=
+begin
+  have hdom : (multiplicity p n).dom,
+  { rw multiplicity.nat.multiplicity_dom_iff, split, exact hn, sorry, },
+  simp [factorization, pp],
+  rw part.get_or_else_of_dom (multiplicity p n) hdom,
+  simp,
```

Comment:
````lean
simp [factorization, pp, coe_mk, if_true, @padic_val_nat_def p (fact_iff.mpr pp) n hn]
````

Reply:
Added. There's a way of making a suggestion so that one can automatically commit it by clicking.